Add ADR 12 to the index

ipolonsk opened the pull request:
Closing issues [Write design document on setting up ACM/ArgoCD management cluster](https://github.com/open-infrastructure-labs/ops-issues/issues/7 ). 
The document describes the new infra cluster high-level design and the process of choosing the installation method.

ipolonsk committed:
Add ADR 12 to the index

sesheta commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **tumido** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @tumido` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fblueprint).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/operate-first/blueprint/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["tumido"]} -->

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+
+## CURRENT STATUS
+
+Our current status consisted of three separate OpenShift clusters one for running development tasks, one for Infrastructure testing and a single (all in one) node cluster.
```
"We currently have three separate OpenShift clusters: one for ..."

larsks replied:
"single node (all in one)" rather than "single (all in one) node"

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+
+### Details about the current clusters
+
+*   Lars’s cluster - A 6 nodes openshift cluster that runs Argocd to deploy application and configuration.
```
The cluster has *9* nodes. Probably worth noting that it is running OCS for storage.

anishasthana replied:
`openshift` - > `OpenShift`

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+
+*   In this solution we are going to use only products that are provided by Red Hat or will be provided by Red Hat in the near future.
+*   We are using a BM only environment which resides in MOC (mass open cloud) Datacenter.
+*   All our deployment steps will be documented.
```
Active voice is better than passive voice: "We will document all our deployment steps"

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+
+## TOOLS
+
+There are two tools to accomplish our desire infrastructure:
```
s/desire/desired/

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+There are two tools to accomplish our desire infrastructure:
+
+*   Using the cloud.redhat.com tool to deploy the cluster
+*   Using the Openshift install binary (KNI) to deploy the cluster (IPI).
```
I think there is an actual name for this product (maybe it's the openshift baremetal installer?). We should check that and use the proper terminology here.

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+### Pros and Cons for cloud.redhat.com (Assisted Installer):
+**Pros:**
+
+- Assisted Installer is a growing RedHat tool to install a new cluster, getting family with it will have future benefits.
```
s/family/familiar/

anishasthana replied:
RedHat -> Red Hat

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+**Pros:**
+
+- Assisted Installer is a growing RedHat tool to install a new cluster, getting family with it will have future benefits.
+- This tool will be part of the ACM and will be used in deploying new clusters via the ACM.
```
s/the ACM/ACM/

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+- Can be easily automated, use API calls.
+
+**Cons:**
+- A dev preview tool, not completed product
```
s/dev preview/tech preview/

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+
+**Cons:**
+- A dev preview tool, not completed product
+- This method includes booting from an image a BM, a more complicated way.
```
I'm not entirely sure what you're trying to say here. Maybe:

> This method requires booting baremetal nodes from an ISO image

I'm not sure this is actually a "con", arguably, booting from virtual media is easier than setting up the PXE server required by the baremetal installer.

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+
+
+
+*   **Openshift infra cluster -** a 3 Bare-metal nodes cluster is our main component.
```
"A three-node baremetal cluster is our main component"

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+
+
+*   **Openshift infra cluster -** a 3 Bare-metal nodes cluster is our main component.
+*   **Managed Cluster -** An OpenSHift cluster that is managed by the hub cluster. The hub will control the updates and resources on this cluster alongside application deployment.
```
s/OpenSHift/OpenShift/

larsks replied:
s/alongside/as well as/

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+
+
+### Cluster Infra Overview
+
```
This section seems less like a "cluster infra overview" and more like a glossary (a list of definitions of words used in the document).

larsks commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+
+### User management
+
+Both Applications and OpenShift will use OpenID Connect provider as the user database. Each permission will be “based on a group” approach.
```
Instead of:

> Each permission will be “based on a group” approach.

Try instead:

> Authorization will be managed through group membership.

tumido reviewed (changes requested):
Can you please re-format this document so it follows the template:
https://github.com/operate-first/blueprint/blob/main/docs/adr/template.md

The whole structure and purpose is a bit different, it's should not be a status report, but an architectural decision report.
https://adr.github.io/

That means:
- The audience, is given
- Revisions are not important since this document is immutable therefore a "new revision" should be tracked as a new ADR
- User management is not subject of this ADR
- Cluster Infra Overview is not the subject either
- Tools are the considered options
etc.

anishasthana commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+## ABOUT THIS DOCUMENT
+
+This document is a High-level Design for the new MOC infra management cluster. This hub cluster will be the central openshift cluster to manage other clusters on-demand using infrastructure as a code to deploy configuration. This centralized cluster will be part of Operate First with RedHat products, testing new dev preview capabilities and providing feedback.
```
`infra` -> `infrastructure`
`openshift` -> `OpenShift`
`RedHat` -> `Red Hat`

anishasthana commented on docs/adr/0012-high-level-design-infra-cluster.md:
```diff
@@ -0,0 +1,119 @@
+### Details about the current clusters
+
+*   Lars’s cluster - A 6 nodes openshift cluster that runs Argocd to deploy application and configuration.
+*   Ilana’s cluster - a 4 node BM cluster that serves for the ‘CNV on MOC’ initiative and was installed using the IPI installer.
```
`BM` -> `Bare Metal`

anishasthana replied:
on a similar note, random people may not know what CNV, MOC, and IPI stand for.

anishasthana reviewed (changes requested):
Requested a bunch of nit changes. Could you make sure to update all instances of `openshift/Openshift/similar` to `OpenShift`? I didn't add the latter few ones.

sesheta commented:
<details>
<summary>Pre-Commit Test failed! Click here</summary>

```
[INFO] Initializing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Initializing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Installing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Tabs remover.............................................................Passed
Trim Trailing Whitespace.................................................Failed
- hook id: trailing-whitespace
- exit code: 1
- files were modified by this hook

Fixing docs/adr/0012-cluster-deployment-methods.md

Check for merge conflicts................................................Passed
Fix End of Files.........................................................Passed
Check for added large files..............................................Passed
Check for case conflicts.................................................Passed
Check JSON...........................................(no files to check)Skipped
Check for broken symlinks............................(no files to check)Skipped
Detect Private Key.......................................................Passed
yamllint.................................................................Passed
```
</details>

tumido commented:
FTR: @ipolonsk rewrote the document so it better follows the ADR template. I think it capturing the architectural decision much  
 better this way. The detailed description should be hosted somewhere else. (I think this is much better choice then my `operators` ADR for example, which to my today's eyes is too detailed.)

Please review once more everybody and let @ipolonsk  know what format you prefer, if there's anything you miss and such. If you prefer the previous format of this PR, please blame me instead. :smile:

anishasthana reviewed (approved):
/lgtm

sesheta commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/operate-first/blueprint/pull/34#pullrequestreview-602017465" title="LGTM">anishasthana</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **tumido** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @tumido` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fblueprint).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/operate-first/blueprint/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["tumido"]} -->

anishasthana commented:
I like the new format a lot better! @ipolonsk can you please squash your commits?

tumido reviewed (commented):
Couple of questions:

1) What is the problem statement? Cluster deployment in general or rather a preferred deployment model for the moc-infra cluster? Or Baremetal cluster deployment in general? Because since we have moc-infra with ACM available I was under impression that we would like to use ACM to deploy new clusters. Can we specify that?
2) Infrastructure as a code is nice, though we probably want to get rid of Ansible playbooks and favor operators/kustomize type of automation. From the content of this ADR it's unclear what kind of infra as a code you mean.
3) Does the IPI/UPI installed cluster offer an easy way to be connected to our management cluster's ACM? (So we can have a single pane of glass for cluster management,) Can you please briefly mention it as a Good/Bad aspect of the options?

I would personally favor the Assisted Installer, though I'm not that familiar with the issues you've faced with it. Can you please elaborate that a bit (or provide some links)?

HumairAK commented on docs/adr/0012-cluster-deployment-methods.md:
```diff
@@ -0,0 +1,41 @@
+
+## Considered Options
+
+### UPI - User-provioned indrastructure
```
` User-provioned indrastructure` -> `User provisioned infrastructure`

HumairAK commented:
This format is fine, I do think that some additional links would be helpful for further reading.

ipolonsk commented:
@tumido, few questions regards your comment:
1. I added more relevance to the moc-infra content, so it can be clear that we are talking only about the moc-infra cluster.
2. I didn't want to mention different names of tools as it can be a subject for another ADR (decide on the automation tools), I tried to keep it simple without explicitly saying the implantation method for other subjects. 
3. Not sure about this one. 
   How the installation method is related to the ACM (that runs on top of that cluster), and how can it be a good/bad option?  After 
   the cluster is deployed with X metod, the single pane of glass can be created independently. It doesn't affect the installation 
   method.
 
We used a tech preview version of the assisted installer to deploy a single node on a disconnected environment and did a massive setup pre-installation.
So we don't have enough information about the assisted installer three-node cluster for this case of deployment. Not sure what links I can add to this doc.

tumido commented:
Hi @ipolonsk sorry to stall this for that long.

> 1. I added more relevance to the moc-infra content, so it can be clear that we are talking only about the moc-infra cluster.

:+1: Can you please push those changes? I don't see this in the PR.

> 2. I didn't want to mention different names of tools as it can be a subject for another ADR (decide on the automation tools), I tried to keep it simple without explicitly saying the implantation method for other subjects.

Ok, that's a fair point.

> 3. Not sure about this one.
>    How the installation method is related to the ACM (that runs on top of that cluster), and how can it be a good/bad option?  After
>    the cluster is deployed with X metod, the single pane of glass can be created independently. It doesn't affect the installation
>    method.

A "Good thing" is that when you use ACM to provision a cluster, it's already connected to ACM.
A "Bad thing" can be that while it may be possible to import a cluster provisioned via a different method to ACM, it probably may requires some expertise in importing/setting things up etc.

It's an aspect to consider when deciding which method to choose - how easy the path forward with managing the cluster is after install. It doesn't effect the installation directly though it matters because if you choose a method that complicates things for the next step, it may not be that good of an option. Just one aspect of many that should not be omitted.

> We used a tech preview version of the assisted installer to deploy a single node on a disconnected environment and did a massive setup pre-installation.
> So we don't have enough information about the assisted installer three-node cluster for this case of deployment. Not sure what links I can add to this doc.

I think some basic documentation links on each of the method would be sufficient. So people know what do you mean by those terms and don't have to google each method. It doesn't have to be precisely a step by step guide but rather links to docs that can give people idea on how each approach works, looks like and what effort has to be put into each.

sesheta commented:
@ipolonsk: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

sesheta commented:
New changes are detected. LGTM label has been removed.

ipolonsk commented:
Hey @tumido
I added few changes to the documentation I made it more relevant for the moc infra since this doc discusses the option for deploying only that initial cluster in the environment.
Since this document describes this situation I don't think the ACM is a relevant point for pro and cons, because we don't have an ACM at the time of setting it up, this is more for the ground zero cluster.
I added useful links to more details for readers. 
Please let me know what do you think (:

tumido commented on docs/adr/0012-moc-infra-cluster-deployment-methods.md:
```diff
@@ -0,0 +1,47 @@
+
+## Context and Problem Statement
+
+This document suggests options for the preferred deployment method for the **moc-infra cluster** at the MOC(Mass Open Cloud). This hub cluster will be the central openshift cluster to manage and deploy other clusters on-demand.
```
```suggestion
This document suggests options for the preferred deployment method for the **moc-infra cluster** at the MOC(Mass Open Cloud). This hub cluster will be the central OpenShift cluster to manage and deploy other clusters on-demand.
```

anishasthana replied:
`openshift` -> `OpenShift`

tumido commented on docs/adr/0012-moc-infra-cluster-deployment-methods.md:
```diff
@@ -0,0 +1,47 @@
+
+This document suggests options for the preferred deployment method for the **moc-infra cluster** at the MOC(Mass Open Cloud). This hub cluster will be the central openshift cluster to manage and deploy other clusters on-demand.
+
+The moc-infra cluster is a bare-metal cluster, deploying an OpenShift Container Platform on bare metal can be achieved by several methodologies.
```
```suggestion
The moc-infra cluster is a bare-metal cluster, deploying an OpenShift Container Platform on bare metal can be achieved by several methods.
```

larsks replied:
This should be two clauses:

```suggestion
The moc-infra cluster is a bare-metal cluster. Deploying an OpenShift Container Platform on bare metal can be achieved by several methods.
```

tumido reviewed (commented):
2 little nitpicks. Otherwise it captures the decision preciselly. I love this edit! @ipolonsk this is good content! :+1:

larsks commented on docs/adr/0012-moc-infra-cluster-deployment-methods.md:
```diff
@@ -0,0 +1,47 @@
+
+Deploy an OpenShift cluster on infrastructure that you prepare and maintain.
+
+We decided not to take this option under consideration since this method includes heavier infrastructure configuration.
```
I think the adjective "heavier" here is a little unclear. Maybe:

```suggestion
We decided not to take this option under consideration since there is more effort involved in setting up the infrastructure.
```

larsks commented on docs/adr/0012-moc-infra-cluster-deployment-methods.md:
```diff
@@ -0,0 +1,47 @@
+* Good, because this tool will be part of ACM deployment of new clusters.
+* Good, because it's a good opportunity to provided feedback to the developers.
+* Good, because can be easily automated, use API calls.
+* Good, because this method requires booting bare-metal nodes from an ISO image.
```
Saying "requires" here makes this sound less like a "good" point and more like a  limitation. Maybe "allows" rather than "requires"?

anishasthana reviewed (changes requested):
One nit

sesheta commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **tumido** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @tumido` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fblueprint).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/operate-first/blueprint/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["tumido"]} -->

tumido reviewed (commented):
@ipolonsk can you please rebase and fix the nit from @anishasthana ? So we can merge it.

Looks great to me otherwise :+1:

tumido reviewed (commented):
/lgtm

HumairAK reviewed (approved):
/lgtm

sesheta commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/operate-first/blueprint/pull/34#pullrequestreview-621364380" title="LGTM">HumairAK</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fblueprint).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/operate-first/blueprint/blob/main/OWNERS)~~ [HumairAK]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
